### DISCUSSION TITLE
diff two open windows

### DISCUSSION THREAD

**Discussion by pr3c0g** [kovidgoyal/kitty#4264] [Q&A] [answered]
Hi all!

First of all, I'm sorry if I'm asking something which is already documented, but I couldn't find it.

I would like to diff two windows.
The use case is, for example, ssh'ing to two different systems to troubleshoot something, and end up checking a configuration file which is present in both of the systems.
By opening that file with VIM (for example), I have both files side by side in my kitty terminal, one in each window.

Is there then a way to diff these two?

Thank you very much in advance!

**Comment by kovidgoyal** [comment, marked as answer]
No there is no diff two windows function, though you can probably easily
write a kitten to accomplish it. Or use the
https://sw.kovidgoyal.net/kitty/kittens/remote_file/ kitten to edit both
files locally and then use whatever local tool you like for diffing.

**Reply by pr3c0g** [reply]
Thank you for your reply. I'm using the scrollback buffer to copy to local vim for now, but I'll try to make that kitten work for me.

**Reply by allemangD** [reply]
You can also use a path like `ssh:hostname:path` with the [diff kitten](https://sw.kovidgoyal.net/kitty/kittens/diff/#command-line-interface) to compare files directly from ssh.

**Reply by pr3c0g** [reply]
That takes a little bit more work than what I'm doing (maybe) but it's also a good option.
